samples/net/cloud/aws_iot_mqtt/src/main.c

```diff
@@ -0,0 +1,494 @@
+/*
+ * Copyright (c) 2023 Lucas Dietrich <ld.adecy@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "creds/creds.h"
+#include "dhcp.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <poll.h>
+#include <unistd.h>
+
+#include <zephyr/data/json.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/net/dns_resolve.h>
+#include <zephyr/net/mqtt.h>
+#include <zephyr/net/sntp.h>
+#include <zephyr/net/tls_credentials.h>
+#include <zephyr/random/rand32.h>
+
+
+#if defined(CONFIG_MBEDTLS_MEMORY_DEBUG)
+#include <mbedtls/memory_buffer_alloc.h>
+#endif
+
+#include <sys/socket.h>
```

Comment:
It seems to me you should rather include `#include <zephyr/net/socket.h>` given that you use `zsock_*` APIs, not POSIX header. Are posix APIs even needed for this sample?

Reply:
I didn't made the change yet, do you suggest to remove `#include <poll.h>` and `#include <unistd.h>`, then use `zsock_*` variants for poll related stuff ? It that case, we could almost disable `CONFIG_POSIX_API` (last POSIX function used is `clock_settime` from  `#include <time.h>`).

If we decide to pursue this approach, my suggestion would be to enable `CONFIG_POSIX_CLOCK` to still have the convenient function `clock_settime` through `#include <zephyr/posix/time.h>` rather than `#include <time.h>`.

What are your thoughts on this?

Reply:
I'd just prefer to be consistent with the API usage, if started to use `zsock_*` APIs, just stick to it.

As for clock, enabling `CONFIG_POSIX_CLOCK` only seems fine.